### ISSUE TITLE
Failed to create script domain because of invalid cast exception

### ISSUE THREAD

**Issue by DurtyFree** [crosire/scripthookvdotnet#736]
SHVDN doesn't seem to start up when scripts have some unmanaged dlls dependencies.

```
[14:34:29] [INFO] Created new script domain with v3.0.0.
[14:34:29] [ERROR] Failed to create script domain':
System.InvalidCastException: Unable to cast transparent proxy to type 'GTA.ScriptDomain'.
   at GTA.ScriptDomain.Load(String path) in H:\DevProjects\GIT_Projects\MTAV\scripthookvdotnet3\source\core\ScriptDomain.cs:line 126
```

Properly resolving assemblies on startup by subscribing the OnResolveAssembly event seems to fix this issue.

```CSharp
        private static Assembly OnResolveAssembly(object sender, ResolveEventArgs args)
        {
            try
            {
                return Assembly.Load(args.Name);
            }
            catch
            {
                return Assembly.LoadFrom(Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location) + "\\" + args.Name.Split(',')[0].Trim() + ".dll");
            }
        }
```

**Comment by Anfit5** [comment]
And where it is necessary to insert this function?

**Comment by crosire** [comment]
Hopefully fixed now as of 4924888f12248a849a8571278b709e3e6596096f.
